easybuild/easyblocks/g/gromacs.py

```diff
@@ -492,14 +492,15 @@ def install_step(self):
             if not self.lib_subdir:
                 raise EasyBuildError("Failed to determine lib subdirectory in %s", self.installdir)
 
-            # Reset installopts etc for the benefit of the gmxapi extension
-            self.cfg['installopts'] = self.orig_installopts
-
     def extensions_step(self, fetch=False):
         """ Custom extensions step, only handle extensions after the last iteration round"""
         if self.iter_idx < self.variants_to_build - 1:
             self.log.info("skipping extension step %s", self.iter_idx)
         else:
+            # Reset installopts etc for the benefit of the gmxapi extension
+            self.cfg['install_cmd'] = self.orig_install_cmd
+            self.cfg['build_cmd'] = self.orig_build_cmd
+            self.cfg['installopts'] = self.orig_installopts
```

Comment:
Hmm, this makes me a bit nervous... Why (only) these easyconfig parameters?

Isn't this a sign of a bigger issue (not specific to GROMACS) that should be fixed in framework?

Reply:
Actually I only need to reset installopts, but since I had saved the others too I reset them.
And the reason for saving them is that I change them from their defaults.

The problem with installopts was that it gets set to "install" causing the subsequent extension pip command to come out as:
```
INFO cmd "export GMXTOOLCHAINDIR=/home/a/ake/Public/easybuild/software/GROMACS/2021-fosscuda-2020b/share/cmake/gromacs_mpi &&
 pip install --prefix=/home/a/ake/Public/easybuild/software/GROMACS/2021-fosscuda-2020b install  --verbose  --no-deps  --ignore-installed  --no-build-isolation  ."
```
Note the double "install", i.e. it tried to install the "install" pkg from pypi... causing the sanity to barf...

So, most easyblocks don't internally change any of these... thus no problem. But installopts could potentially be worth keeping an eye on before going into pythonpackage extension installation.